src/apps/filters/h3.c

```diff
@@ -312,6 +312,89 @@ SUBCOMMAND(getIndexDigit,
     return E_SUCCESS;
 }
 
+SUBCOMMAND(constructCell,
+           "Construct an H3 cell from resolution, base cell, and digits") {
+    DEFINE_FORMAT_ARG(
+        "'json' for \"CELL\"\\n, 'newline' for CELL\\n (Default: json)");
+    int res = 0;
+    int bc = 0;
+    char digitsStr[1024] = {0};
+
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText = "Resolution, 0-15 inclusive."};
+    Arg bcArg = {.names = {"-b", "--baseCell"},
+                 .required = true,
+                 .scanFormat = "%d",
+                 .valueName = "base",
+                 .value = &bc,
+                 .helpText = "Base cell number, 0-121 inclusive."};
+    Arg digitsArg = {.names = {"-d", "--digits"},
+                     .required = false,
+                     .scanFormat = "%1023c",
+                     .valueName = "d0,d1,...",
+                     .value = &digitsStr,
+                     .helpText =
+                         "Comma-separated list of child digits (0-6) of length "
+                         "resolution. Ignored if resolution is 0."};
+
+    Arg *args[] = {&constructCellArg, &helpArg,  &resArg, &bcArg,
+                   &digitsArg,        &formatArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+
+    // We expect `res` digits. We add one more to make sure that the user didn't
+    // pass more.
+    int digits[MAX_H3_RES + 1] = {0};
+    if (res > 0) {
+        if (!digitsArg.found) {
+            return E_DIGIT_DOMAIN;
+        }
+
+        int count = 0;
+        char *p = digitsStr;
+        while (*p && count < MAX_H3_RES + 1) {
+            while (*p == ',') {
+                p++;
+            }
+            if (!*p) {
+                break;
+            }
+            char *endptr = NULL;
+            long val = strtol(p, &endptr, 10);
```

Comment:
Why a `long` here when the only valid digits are 0-7 (really 0-6 for the purpose of this function since unused digits aren't required). A`byte`(or `uint8`) should be more than enough.

It could potentially just be a switch on the characters `'0'` to `'6'` with a default of returning an error, and that would likely produce an efficient jump table. (Though there's no need to count cycles for something so small.)